### PR TITLE
Design changes of month view

### LINKED ISSUE #3300: Design of month view
## Is your feature request related to a problem? Please describe.
* The current day is the start of the orientation inside the calendar grid but currently it is hard to find (especially if there are many colored events inside the month view).  
* I can't tell the current month from the days of the previous or next month. 

## Describe the solution you'd like
As a user of the calendar, I want an improved design which emphasizes relevant information, because it helps to get orientaition inside the calendar grid.

## Describe alternatives you've considered
Proposal as mockup: 

The current day (today) is more dominant and better/faster to identify. The darkening of days of the previous and following months emphasize the current month, which is the one of interest.

grafik

Added CSS: 
```
.fc-daygrid-day.fc-day.fc-day-other, .fc .fc-daygrid-day.fc-day-today.fc-day-other {
	background-color: var(--color-background-dark) !important;
}

.fc-day-today .fc-daygrid-day-top {
	background-color: var(--color-primary);
}

```

Comparing current view:
grafik

### PR DESCRIPTION
**Pull request by dartcafe**
resolves #3300
screenshots see linked issue

I had to simulate this, since the installation of the dev setup breakes here

### CODE REVIEW

**Comment by szaimen** [comment]
@dartcafe Thanks for your PR!
If you dev-setup fails, you can still test it by running the following command:
<details>
<summary>click here</summary>

```
docker run -it \
-e CALENDAR_BRANCH=enh/month-view-css \
-p 127.0.0.1:8443:443 \
--name nextcloud-easy-test \
ghcr.io/szaimen/nextcloud-easy-test:latest
```

</details>

**Comment by dartcafe** [comment]
Currently I have no docker installed. But I will try this later. Thanks for the hint.

**Comment by jancborchardt** [comment]
Since it's a design changey can I ask you to add before & after screenshots for easier review? Thank you! :)

**Comment by dartcafe** [comment]
> resolves #3300
> screenshots see linked issue

**Review by st3iny — COMMENTED** [review]
I tested it and it works. The code looks good too.

However, the contrast of the dates outside the current month is very low. This is not good for accessibility. This issue persists when selecting the high contrast theme.

Screenshot 2021-12-17 at 12-22-44 Dezember 2021 - Kalender - calendar-3817-slrcjvqw

**Comment by szaimen** [comment]
> However, the contrast of the dates outside the current month is very low. This is not good for accessibility. This issue persists when selecting the high contrast theme.
> 
> Screenshot 2021-12-17 at 12-22-44 Dezember 2021 - Kalender - calendar-3817-slrcjvqw

Also thought this, too.

Should be solvable with the following CSS:
```css
.fc .fc-day-other .fc-daygrid-day-top {
	opacity: 1;
}
```

**Comment by dartcafe** [comment]
Another glitch maybe is the week number. The height could possibly be adjusted to match the day-top height?

grafik

**Comment by szaimen** [comment]
> Another glitch maybe is the week number. The height could possibly be adjusted to match the day-top height?

What about using the primary color for the whole square (of today)? Would probably make the day much more visible again...

**Comment by dartcafe** [comment]
> Also thought this, too.
> 
> Should be solvable with the following CSS:
> 
> ```css
> .fc .fc-day-other .fc-daygrid-day-top {
> 	opacity: 1;
> }
> ```

This is with opacity: 0.6: 
grafik

**Comment by dartcafe** [comment]
> What about using the primary color for the whole square (of today)?

This could get possibly a way too heavy, depending on the sites main color and we then have to deal with the foreground color. 

Mocked variants:

grafik

**Comment by szaimen** [comment]
> > What about using the primary color for the whole square (of today)?
> 
> This could get possibly a way too heavy, depending on the sites main color and we then have to deal with the foreground color.
> 
> Mocked variants:

And applying it only to today and not to more days would also already be to heavy?

**Comment by szaimen** [comment]
> With adjusted height (2.1em):
> grafik

LGTM if we decide to not go with the full square of today using the primary color.

**Comment by dartcafe** [comment]
> And applying it only to today and not to more days would also already be to heavy?

It should be applied only to today. The other days were made for displaying the possible variants. (i.e. 30th vs. 21th.)

**Comment by szaimen** [comment]
I think giving the not all-day events a background-color woulld probably work best as we don't need to care about the contrast between background and text anymore:
image

**Comment by dartcafe** [comment]
Just another try for the full background of today: 

```
.fc .fc-daygrid-day.fc-day-today,
.fc .fc-timegrid-col.fc-day-today {
 /*! background-color:unset !important */
}

.fc .fc-daygrid-day.fc-day-today {
      background-color: var(--color-primary);
}

.fc-day-today .fc-daygrid-dot-event .fc-event-time, .fc-day-today .fc-daygrid-dot-event .fc-event-title {
	color: var(--color-primary-text);
}
```

Proposal | Current
-|-
grafik | grafik



Any opinions from @nextcloud/designers

**Comment by dartcafe** [comment]
> I think giving the not all-day events a background-color

Also a good idea. Personally, I prefer the current layout, because you can tell the full day events from the appointments with a start date just by the background style.

**Comment by szaimen** [comment]
> Just another try for the full background of today:

Looks already much better 👍

Do you mind adding the following css for a 3rd variant and adding it to your table?

```
.fc-event-today {
    background-color: var(--color-background-dark);
}

.fc .fc-day-other .fc-daygrid-day-top {
	opacity: 1;
}
```

**Comment by jancborchardt** [comment]
Applying the primary color as a full background color is a bit much though. And it results in heavy readability issues. ;)

2 alternatives which can be used in addition to the current primary-color circle around the day:
- Primary color (or primary-light) border around the current day
- color-primary-light background color of the cell (instead of full primary) but even that will lead to issues with readability.

**Comment by szaimen** [comment]
> And it results in heavy readability issues. ;)

Thats solved since the text becomes a background which makes it readable again :)

**Comment by dartcafe** [comment]
More mockups

primary-light | border full | border top
-|-|-
grafik | grafik | grafik

**Comment by szaimen** [comment]
I vote for the primary-light border variant because it is then coherent to the weekview solution :) https://github.com/nextcloud/calendar/pull/3831

**Comment by dartcafe** [comment]
@szaimen You refer to this, right? 
[grafik](https://user-images.githubusercontent.com/26707476/146823528-16f2ec7f-978a-406a-bdae-78a4540d413c.png)

@jancborchardt Your word?

**Comment by szaimen** [comment]
> @szaimen You refer to this, right?

yes 👍

**Comment by dartcafe** [comment]
new Proposal: 
current | **new**
-|-
grafik | grafik | 

Changes: 
```css
.fc-daygrid-day.fc-day.fc-day-other, 
.fc .fc-daygrid-day.fc-day-today.fc-day-other {
    background-color: var(--color-background-dark) !important;
}

.fc-daygrid-day.fc-day.fc-day-other .fc-daygrid-day-top, 
.fc .fc-daygrid-day.fc-day-today.fc-day-other .fc-daygrid-day-top {
    opacity: 0.6;
}

.fc .fc-daygrid-day.fc-day-today,
.fc .fc-timegrid-col.fc-day-today {
    // background-color: unset !important;
    background-color: var(--color-primary-light) !important;
}

```

**Comment by tcitworld** [comment]
Also related this old PR #2521

**Comment by dartcafe** [comment]
@szaimen I cherry-picked your pr and added the last changes. Seems to fit in all views now.

**Comment by miaulalala** [comment]
The background colour is also the same colour as the border of the boxes so it makes the whole row look like a single block. Can this be changed? This is february for me:

image

I find it very heavy.

**Comment by dartcafe** [comment]
Adding borders
grafik

**Comment by miaulalala** [comment]
@jancborchardt can you give one last ok on this before I merge?

File: css/fullcalendar.scss
Comment (szaimen) [inline css/fullcalendar.scss:245]: I wonder if we can revert this now that we applied the styling globally. Question is if the global styling is wanted because it modifies the today view as well? Probably all views should be looked at because of that?

**Review by jancborchardt — APPROVED** [review]
Looks great now! :)

@dartcafe @szaimen small possible enhancement for week and month view is to also give the table header (with the date text) the primary-light background, but that can also be done in a follow-up.

File: css/fullcalendar.scss
Comment (miaulalala) [reply css/fullcalendar.scss:245]: Can you post screenshots of the week / day view please?

**Comment by dartcafe** [comment]
@miaulalala 
week | today | tomorrow
-|-|-
grafik | grafik | grafik

@szaimen Any idea for the horizontal lines?

**Comment by dartcafe** [comment]
> small possible enhancement for week and month view is to also give the table header (with the date text) the primary-light background, but that can also be done in a follow-up.

Would be better to do this in another PR, so I can clean up my desk. 😜

**Comment by miaulalala** [comment]
> @miaulalala
> 
> week	today	tomorrow
> grafik	grafik	grafik

I think that looks fine! @jancborchardt ?

**Comment by jancborchardt** [comment]
Sorry I misspoke, I meant the small enhancement would be nice for the _day_ view: The header cell of "today" could use primary-light background.

But as said: Can be a different PR. All good 👍

File: css/fullcalendar.scss
Comment (szaimen) [reply css/fullcalendar.scss:245]: @dartcafe please delete this code again since @jancborchardt agreed to make it globablly for all views.

```
		.fc-event {
			box-shadow: 0px 0px 0px 1px var(--color-primary-light) !important;
		}
```
and
```
		&.fc-col-header-cell {
			background-color: var(--color-primary-light);
		}
```
is probably still missing from the global styling though...

**Comment by miaulalala** [comment]
@dartcafe would you mind opening a ticket for the suggested changes by Jan? :pray:

**Pull request merged**
